Add Latvian orthophoto 2016–2018 (License: CC0)

soshial opened the pull request:
The link to WMS capabilities [xml info](http://lvmgeoserver.lvm.lv/geoserver/ows?service=wms&version=1.3.0&request=GetCapabilities&layer=public:LIDARHeight2m).

Files and info:
- [This file](https://github.com/osmlab/editor-layer-index/files/5824426/Aerofotograf275353anasd.pdf) has time vs. quadrant information.
- The [official page](https://map.lgia.gov.lv/index.php?lang=0&cPath=4_16&txt_id=49&0&ap=117) with some detailed info about the map

rbuffat commented:
@soshial Thanks for this PR! I had a quick look at it. 

attribution/text is "Latvijas Ģeotelpiskās informācijas aģentūras brīvais WMS serviss 1994 - 1999.gada Latvijas ortofotokarte"

Is this correct? The description mentions the years 2016-2018.  

Also, do you know if the ["Orthophoto 1st cycle (1994-1999) - Latvia" ](https://raw.githubusercontent.com/osmlab/editor-layer-index/gh-pages/sources/europe/lv/Orthophoto1stcycle(1994-1999)-Latvia.geojson) source is still available? The current URL seems not to be reachable anymore.

soshial commented:
1. > attribution/text is "Latvijas Ģeotelpiskās informācijas aģentūras brīvais WMS serviss 1994 - 1999.gada Latvijas ortofotokarte"

    Actually, it was left from ortophoto 1994–1999 geojson file. It's a mistake. What should we write in this field?
2. How do we specify a zoom for this map?
3. How can I extract geojson coverage from [these files](http://map.lgia.gov.lv/index.php?lang=0&cPath=4_16&txt_id=49&0&ap=117)?
4. > Also, do you know if the "Orthophoto 1st cycle (1994-1999) - Latvia" source is still available? The current URL seems not to be reachable anymore.
    I noticed, that it doesn't work. I wrote a comment there. I guess you have already noticed it.

PS. Also, there is a historical map from 1920-1940. I may add it as well. Do we need a license for that?
Should I add current [topographical map of Latvia](https://www.lgia.gov.lv/en/topographic-map-scale-150-000-2nd-edition) from that same state agency (LGIA) at the scale 1:50 000 under CC0 license? Do we need it?

rbuffat commented:
I had the time to have a closer look at it. 

The orthophoto data itself is from the Latvian Geospatial Information Agency and its license is stated as CC0: https://www.lgia.gov.lv/en/color-orthophoto-map-2016-2018-cycle-6

The WMS server is operated by JSC Latvia's State Forests: https://www.lvmgeo.lv/en/data

```
The LVM GEO team is actively involved in the development of geospatial information technologies in Latvia and supports open data initiatives. LVM GEO maintains one of the biggest geospatial data and service collections in Latvia, a portion of those we have published as open data available for any user. We encourage the use of these data and services for educational, scientific, research and software development purposes
```

Under Services -> Orthophoto:

```
JSC Latvia's State Forests offer orthotphoto WMS service that covers whole territory of Latvia. This service has been prepared by processing and combining Latvian Geospatial Information Agency's 6th cycle orthophotos, each area is covered by the latest available orthophoto image. This service is available along other LVM GEO published WMS services:
 
http://lvmgeoserver.lvm.lv/geoserver/ows?service=wms&version=1.3.0&request=GetCapabilities&layer=public:Orto_LKS
 
Service is prepared using: Orthophoto map 1:10 000 data © Latvian Geospatial Information Agency, 2016.- 2018. 
```

The updated wms server url for the already existing source can be found here: https://www.lgia.gov.lv/lv/wms-servisi

```
Ortofoto kartes 1994.-1999. gads ar aerofotografēšanas laika kartoshēmu  – šis serviss ir bez maksas brīvi pieejams
Ortofoto kartes 1994.-1999. gads, šis serviss ir bez maksas brīvi pieejams
```
" brīvi pieejams" links to http://services.lgia.gov.lv/arcfree/services/WMS_Ortofoto_v1/MapServer/WmsServer?service=WMS&REQUEST=GetCapabilities&VERSION=1.3.0&

> 1. Actually, it was left from ortophoto 1994–1999 geojson file. It's a mistake. What should we write in this field?

The attribution text is displayed in on OSM editor when the imagery is shown. I think `LĢIA (Latvian Geospatial Information Agency) provided these open data (CC0). WMS service is provided by LVM (Latvian State Forestry).` is fine. 

> 2. How do we specify a zoom for this map?

There is a tool to find zoom levels: https://mybinder.org/v2/gh/rbuffat/eli-helper/master?urlpath=apps%2F/check_minzoom.ipynb
It is quite new and there can be bugs.

I think either no "min_zoom" or "min_zoom": 1 can be used. 

> 3. How can I extract geojson coverage from these files?

The link shows for me only an empty page. If there are geodata available, a good tool would be [QGIS](https://qgis.org/en/site/). Finding a good geometry is a bit tricky, there are some tradeoffs between an accurate representation of the dataset and the size of the resulting geometry to be made. If you are not unhappy with the geometry of the old source you can just use this. 

> PS. Also, there is a historical map from 1920-1940. I may add it as well. Do we need a license for that?

Ideally, for each source contained in ELI the license is known and compatible with OSM. Some editors filter out old imagery.  E.g. iD filters out datasets that are older than 20 or 30 years.

> Should I add current topographical map of Latvia from that same state agency (LGIA) at the scale 1:50 000 under CC0 license? Do we need it?

If you think that a dataset is valuable and the license is compatible, sure.

rbuffat commented:
@soshial there is a bug that failed the tests for your changes in the most recent commits. Would it be possible for you to merge the current gh-pages branch in your PR?

soshial commented:
Let's wait, when the old website starts working. But I will put some additional files in this post:

* 7th cycle geotemporal map: [7cikla20192020gadsor.pdf](https://github.com/osmlab/editor-layer-index/files/5825481/7cikla20192020gadsor.pdf)
* 6th cycle geotemporal map 1: [Aerofotograf275353anasd.pdf](https://github.com/osmlab/editor-layer-index/files/5824426/Aerofotograf275353anasd.pdf)
* 6th cycle geotemporal map 2: [Inform257cijapar6ciklaa-2.pdf](https://github.com/osmlab/editor-layer-index/files/5825483/Inform257cijapar6ciklaa-2.pdf)
* 1th cycle geotemporal map: [Ortofoto_1cikls.pdf](https://github.com/osmlab/editor-layer-index/files/5825484/Ortofoto_1cikls.pdf)
* 6-th cycle ArcGIS files: [Inform257cijapar6ciklaa.zip](https://github.com/osmlab/editor-layer-index/files/5825490/Inform257cijapar6ciklaa.zip)
  <img src="https://user-images.githubusercontent.com/100644/104829854-93e9b000-5891-11eb-83a5-293593d43334.png" width="250px"/>
* 6-th cycle DGN file: [TOPO5_nomenklatura_dgn.zip](https://github.com/osmlab/editor-layer-index/files/5825492/TOPO5_nomenklatura_dgn.zip)

Is it possible to add these files to the folder with geojsons?

soshial commented:
1. I rebased it, but the check still doesn't pass.
2. I don't think that LĢIA has `privacy_policy_url` for (orthophoto_1999). Can we omit it?
3. Another website uses these parameters for orthophoto 2016: `https://lvmgeoserver.lvm.lv/geoserver/ows?=&service=WMS&request=GetMap&layers=public:Orto_LKS&styles=&format=image/jpeg&transparent=false&version=1.1.1&isOverlay=false&code=F&scaleDependent=true&print=true&jnx=false&height=256&width=256&srs=EPSG:3857&bbox={bbox}`. Do we need some of these params?
4. I added license URL, but I couldn't find a designated field for standard licenses (e.g.: `CC0`, `CC-BY` etc.). Is it possible to include this one? I'd also like to include a contact person email.
5. I would also like to include a link to source [GeoJSON ](https://s3.storage.pub.lvdc.gov.lv/lgia-opendata/citi/1_50000_v2/1_50000_v2_geojson.7z) or [ArcGIS](https://s3.storage.pub.lvdc.gov.lv/lgia-opendata/citi/1_50000_v2/1_50000_v2_arcgis_file_db.zip) files for the [topographic map](https://www.lgia.gov.lv/en/topographic-map-scale-150-000-2nd-edition) I am planning to add in the next commit.

rbuffat commented on sources/europe/lv/lv_orthophoto_cycle1_1994-1999.geojson:
```diff
@@ -2,7 +2,7 @@
-        "name": "Orthophoto 1st cycle (1994-1999) - Latvia",
+        "name": "Orthophoto (1994–1999), B&W, 1:30000, Latvia",
         "type": "wms",
         "url": "https://services.lgia.gov.lv/arcfree/services/Ortofoto_v1/MapServer/WmsServer?LAYERS=0&STYLES=&FORMAT=image/jpeg&CRS={proj}&WIDTH={width}&HEIGHT={height}&BBOX={bbox}&VERSION=1.3.0&SERVICE=WMS&REQUEST=GetMap",
```
Would it possible to change the url to "https://services.lgia.gov.lv/arcfree/services/WMS_Ortofoto_v1/MapServer/WmsServer?..."

soshial replied:
As long as you're sure that it works. I have no idea, how to check that WMS service works. Have you checked the 6th cycle orthophoto?

rbuffat replied:
In most cases the checks are enough, but in this case there seems to be an issue with the WMS server. The server does not follow the WMS 1.3.0 specification and does not invert the x,y coordinates for the bbox parameter for the EPSG:4326 coordinate system.
 
The following parameters work:

```
 "url": "https://lvmgeoserver.lvm.lv/geoserver/ows?width={width}&height={height}&bbox={bbox}&format=image/vnd.jpeg-png8&request=GetMap&service=WMS&styles=&transparent=true&version=1.1.1&srs={proj}&layers=public:Orto_LKS",
```
  
```     
"available_projections": [
            "CRS:84",
            "EPSG:3059",
            "EPSG:3857",
            "EPSG:4326"
        ],
```

soshial replied:
I will change at once to the values you wrote, correct?

soshial replied:
Done.

rbuffat commented:
> 1. I rebased it, but the check still doesn't pass.

Thanks!

> 2. I don't think that LĢIA has `privacy_policy_url` for (orthophoto_1999). Can we omit it?

Yes. The test will fail because currently there is not yet a value for "we have checked for a privacy policy but we have not found one" yet, but this is ok. 

> 3. Another website uses these parameters for orthophoto 2016: `https://lvmgeoserver.lvm.lv/geoserver/ows?=&service=WMS&request=GetMap&layers=public:Orto_LKS&styles=&format=image/jpeg&transparent=false&version=1.1.1&isOverlay=false&code=F&scaleDependent=true&print=true&jnx=false&height=256&width=256&srs=EPSG:3857&bbox={bbox}`. Do we need some of these params?

The parameters in your url look ok. I don't think more are needed. 

> 4. I added license URL, but I couldn't find a designated field for standard licenses (e.g.: `CC0`, `CC-BY` etc.). Is it possible to include this one? I'd also like to include a contact person email.

It is enough to link to the page where the license is mentioned. 

> 5. I would also like to include a link to source [GeoJSON ](https://s3.storage.pub.lvdc.gov.lv/lgia-opendata/citi/1_50000_v2/1_50000_v2_geojson.7z) or [ArcGIS](https://s3.storage.pub.lvdc.gov.lv/lgia-opendata/citi/1_50000_v2/1_50000_v2_arcgis_file_db.zip) files for the [topographic map](https://www.lgia.gov.lv/en/topographic-map-scale-150-000-2nd-edition) I am planning to add in the next commit.

You could mention in the description that the geodata for the map can be downloaded from the homepage.

soshial commented:
> You could mention in the description that the geodata for the map can be downloaded from the homepage.

Can I use Markdown or HTML tags there?

rbuffat commented:
> Let's wait, when the old website starts working. But I will put some additional files in this post:
> 
> * 7th cycle geotemporal map: [7cikla20192020gadsor.pdf](https://github.com/osmlab/editor-layer-index/files/5825481/7cikla20192020gadsor.pdf)
> * 6th cycle geotemporal map 1: [Aerofotograf275353anasd.pdf](https://github.com/osmlab/editor-layer-index/files/5824426/Aerofotograf275353anasd.pdf)
> * 6th cycle geotemporal map 2: [Inform257cijapar6ciklaa-2.pdf](https://github.com/osmlab/editor-layer-index/files/5825483/Inform257cijapar6ciklaa-2.pdf)
> * 1th cycle geotemporal map: [Ortofoto_1cikls.pdf](https://github.com/osmlab/editor-layer-index/files/5825484/Ortofoto_1cikls.pdf)
> * 6-th cycle ArcGIS files: [Inform257cijapar6ciklaa.zip](https://github.com/osmlab/editor-layer-index/files/5825490/Inform257cijapar6ciklaa.zip)
>   <img alt="" width="250px" src="https://user-images.githubusercontent.com/100644/104829854-93e9b000-5891-11eb-83a5-293593d43334.png">
> * 6-th cycle DGN file: [TOPO5_nomenklatura_dgn.zip](https://github.com/osmlab/editor-layer-index/files/5825492/TOPO5_nomenklatura_dgn.zip)
> 
> Is it possible to add these files to the folder with geojsons?

I think the files itself are not needed. The important part is to have a geometry that approximates the imagery.

> > You could mention in the description that the geodata for the map can be downloaded from the homepage.
> 
> Can I use Markdown or HTML tags there?

I don't think that OSM editors support markdown. So I would avoid it.

soshial commented:
I am sorry for so many separate commits. I think I am done with these two maps :)

soshial commented:
Anything I can do to finish this PR? When would they appear in the OSM editor?

PS. I am looking for some open data Georgian maps ([available here](http://maps.tbilisi.gov.ge)).
1. How can I check license and features for this URL? `https://nv.napr.gov.ge/geoserver/wms?LAYERS=NG_REG_LAYER&STYLES=&TRANSPARENT=TRUE&TILED=true&TILESORIGIN=41.0%2C41.0&ENV=fill_opacity%3A0%3B&SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&FORMAT=image%2Fpng&SRS=EPSG%3A900913&BBOX=4974800.4222765,5114311.5618492,4974953.296333,5114464.4359058&WIDTH=256&HEIGHT=256`
2. I also found this [list of layers](https://nv.napr.gov.ge/geoserver/web/wicket/bookmarkable/org.geoserver.web.demo.MapPreviewPage?0&filter=false). I have a feeling that this wasn't supposed to be public list. But maybe it is!

rbuffat commented:
> Anything I can do to finish this PR? When would they appear in the OSM editor?

Thanks for the patience! It looks good to me. If you use iD currently the imagery index is only updated with each release. This can take some weeks in some cases.

If you use a "Unix" based system it is quite easy to setup iD locally:

You have to check out the current editor-layer-index and download the latest release of ELI from https://github.com/openstreetmap/iD

in the editor-layer-index directory you can run:
``` 
make clean
make
```

An in the iD directory execute the following commands:
```
rm -rf node_modules/editor-layer-index/
cp -r path/to/editor-layer-index/ node_modules/
npm install
npm run imagery
npm run all
npm start
```

> PS. I am looking for some open data Georgian maps (available here).
> 
> How can I check license and features for this URL? 

Normally there is somewhere that metadata can be found for a dataset. If no information is found it has to be assumed that the data cannot be used. But it might be worth to try to find a contact person and ask if they would give permission to use the data. https://github.com/osmlab/editor-layer-index/blob/gh-pages/FAQ.md#what-imagery-licenses-are-compatible-with-this-index. If they are willing to sign [a waiver](https://wiki.osmfoundation.org/wiki/Licence/Waiver_and_Permission_Templates) the data can be used.
